pkg/endpoint/restore.go

```diff
@@ -132,7 +131,13 @@ func (e *Endpoint) RegenerateAfterRestore() error {
 	return nil
 }
 
-func (e *Endpoint) restoreIdentity() error {
+type identityAllocator interface {
+	WaitForInitialGlobalIdentities(context.Context) error
+	AllocateIdentity(context.Context, labels.Labels, bool) (*identity.Identity, bool, error)
+	Release(context.Context, *identity.Identity) (released bool, err error)
+}
```

Comment:
Maybe this interface should be defined in `pkg/identity` instead of defining it multiple times in different packages?

Reply:
+1 (either there or in its own package if that helps), also please add the comments for expected behaviour of implementations of this interface.

Reply:
Will be fixed in next upload!